torch/distributed/device_mesh.py

```diff
@@ -212,7 +212,7 @@ def __init__(
             if isinstance(mesh, torch.Tensor) and mesh.device.type != "cpu":
```

Comment:
We already check that the `mesh` tensor is on CPU. There is no need to include the `device="cpu"` in the `.to()` call.

Reply:
LGTM